### PR TITLE
Export FindTenantID function

### PR DESCRIPTION
**Pull request by cmurphy**
rancher/rancher will use this same function to look up a cloud
credential's tenant ID if one is not provided.

https://github.com/rancher/rancher/issues/33097

### CODE REVIEW

**Review by rmweir — APPROVED** [review]
LGTM

File: drivers/azure/azureutil/azureutil.go [hunk 1]
```diff
@@ -541,7 +541,7 @@ func (a AzureClient) removeOSDiskBlob(ctx context.Context, resourceGroup, vmName
 		log.Debugf("Did not find container in storage account")
 	}
 
-	lcrp, err := blobContainersClient.List(ctx, resourceGroup, storageAccount, "", "")
+	lcrp, err := blobContainersClient.List(ctx, resourceGroup, storageAccount, "", "", "")
```
Comment (cbron) [inline drivers/azure/azureutil/azureutil.go:544]: Assuming this is just from SDK bump

**Review by cbron — APPROVED** [review]
lgtm if we scan sdk version changes first.

File: drivers/azure/azureutil/azureutil.go (hunk 1 above)
Comment (cmurphy) [reply drivers/azure/azureutil/azureutil.go:544]: Yes, the List function signature changed

**Comment by cmurphy** [comment]
The breaking changes are noted in https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md but it's a little hard to digest, but I'm confident that breaking changes would be caught by the compiler and there was only one change needed to one function call.

**Pull request merged**